packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -239,12 +240,25 @@ export class OdspDocumentService implements IDocumentService {
                 ? Promise.resolve(null)
                 : this.getWebsocketToken!(options);
 
+            // Annotates the error with the given step and then rethrows, preserving canRetry and retryAfterSeconds
+            const annotateConnectionError = (step: string) => (error: any) => {
+                const normalizedError = normalizeError(error, { props: {
+                    failedStep: step,
+                }});
+
+                // We need to preserve these properties which are used in a non-typesafe way throughout driver code
+                const { canRetry, retryAfterSeconds } = error;
```

Comment:
Is this needed mostly for errors thrown from outside of our repo? I'd assume that all errors would have all the proper fields for normalizeError() to be noop.
If that's the case, should this kind of operation happen closer to where error enters our repo?

Reply:
This is out of paranoia.  See #8743 which would clean this up throughout and ensure this kind of thing isn't necessary.

I did a quick scan of the code for `canRetry` and `retryAfterSeconds` and we're probably ok. But for example the error below is raised from summaryGenerator.ts, and `retryAfterSeconds` would be lost by `normalizeError` because `new LoggingError` doesn't get you an `errorType` which is a requirement for `normalizeError` to noop.

```
const error = new LoggingError(`Received summaryNack: ${message}`, { retryAfterSeconds });
```

Really anywhere that takes an error as `any` and puts stuff on there is at risk of losing that whenever we call `normalizeError`, and of course I can't enumerate all these specific cases here.  In practice, you are right that we will probably get one of our own errors here and all will be well.  I got burned losing `canRetry` in the token provider callback so erring on the side of caution, but if this stinks too much to you I can remove it.

Reply:
Got it. While I do not like it that much, but if we are paranoid, I'd rather move it to normalizeError(). While it's wrong from layering perspective (normalizeError should not even know about such things), I think it's better to have it in one place vs. having similar code across code base (as I'd guess we will have a need for similar wrapping in other places, sooner or later).

Reply:
Ok I can do that, good point